enhancements/network/on-prem-bgp-networking.md

```diff
@@ -0,0 +1,444 @@
+Also, it is already used as a [backend for MetalLB](https://github.com/metallb/metallb/blob/38e16c385716464c140f359f7821db165262e329/design/0001-frr.md).
+
+Alternatives:
+- MetalLB: as it was discussed [here](https://github.com/metallb/metallb/issues/168#issuecomment-1171100455), the implementation would be very complex and not really worth it.
+  For the API VIP, we currently have a chicken and egg problem where we need to have the API VIP up to be able to deploy and run MetalLB. However, Ingress VIPs could potentially be moved (on day 2 by the customer) to a MetalLB deployment.
+- [kube-vip as static pod](https://kube-vip.io/docs/installation/static/): this option has been evaluated and tested. The project is very interesting but we would find ourselves quickly limited by its feature set. For example, kube-vip doesn't have support for BFD, so we would need to deploy it ourselves.
+
+#### Virtual IP addresses
+
+VIPs (Virtual IPs) are used to provide failover of the services across the relevant machines
+(including the bootstrap instance).
+Two control plane services need VIPs:
+- API
+- Ingress
+
+##### API VIPs
+
+The API VIPs are used for API communication and they are provided by the user
+via the `install-config.yaml` [parameter](https://github.com/openshift/installer/blob/8fdc6e43889b111aeb233e5597d1f03d42bb297a/pkg/types/baremetal/platform.go#L177-L185)
+or `openshift-installer` terminal prompts.
+
+The API VIPs can be private or public IPv4/IPv6 addresses
+The external `api.$cluster_name.$base-domain` DNS record should point to API VIPs.
+
+##### Ingress VIPs
+
+The Ingress VIPs are used for Ingress access and they are also provided by the user via the `install-config.yaml` [parameter](https://github.com/openshift/installer/blob/5f7fe178c007dce006d2d59800a5685a80591ede/pkg/types/baremetal/platform.go#L194-L201)
+or `openshift-installer` terminal prompts.
+
+The Ingress VIPs can be private or public IPv4/IPv6 addresses, the external wildcard `*.apps.$cluster_name.$base-domain` DNS record should point to Ingress VIPs.
+
+### Highly available routed VIPs
+enhancements/installer/vsphere-ipi-zonal.md
+Access to the Kubernetes API (port 6443) and Ingress from clients both external
+and internal to the cluster should be highly available across control plane machines.
+
+The API VIPs first resides on the bootstrap instance and then move to one of the control plane nodes, after the bootstrap process has completed and the VIPs have been removed,
+to ensure that the API on the control plane is fully functional before the switch.
+
+The Ingress VIPs will be created on the control plane nodes where the Ingress Controller reports healthy.
+
+FRR will be deployed as a [static pod](https://kubernetes.io/docs/tasks/administer-cluster/static-pod/) and the assets will be rendered by the Machine Config Operator.
```

Comment:
Is there anything about this deployment that has a consensus based algorithm, eg does it have some leader/follower semantic? Do we need to be precious about these pods when we replace control plane machines?

Reply:
Ok that sounds good, might be worth making a note that no consideration needs to be made when replacing control plane machines for the stability of the system, just to rule out any questions about interactions with CPMS